hack/vendor.sh

```diff
@@ -19,7 +19,12 @@ case $# in
 	;;
 # If user passed arguments to the script
 1)
-	eval "$(grep -E "^clone [^ ]+ $1" "$0")"
+	path=$(pwd)"/hack/vendor.sh"
+	if ! cloneGrep="$(grep -E "^clone [^ ]+ $1" $path)"; then
+    	echo >&2 "error: failed to find 'clone ... $1' in $0"
+    	exit 1
```

Comment:
"$BASH_SOURCE" will always refer to the current file in Bash, regardless of
how the script was invoked (even if it was "source"d from another file).  I
think the issue you describe is the difference between "$0" and
"$BASH_SOURCE" -- "$0" is the command/file which the user invoked, which
may not be the same as the path to the script.